lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1443,9 +1443,17 @@ DebugLoc SPIRVToLLVMDbgTran::transDebugScope(const SPIRVInstruction *Inst) {
   unsigned Col = 0;
   MDNode *Scope = nullptr;
   MDNode *InlinedAt = nullptr;
+
+  assert(!(Inst->getLine() && Inst->getDebugLine()) && "OpLine and DebugLine must not both be used");
```

Comment:
TBH I doubt that - there is no such rule in the spec.

Reply:
What should be done if we have SPIRV that has both.  The spec does not show how to resolve a collision:

DebugLine
Specify source-level line and column information. This information applies to all following instructions, up to the first occurrence of any of the following: **the next end of block, the next DebugLine instruction, or the next DebugNoLine instruction**.

OpLine
Add source-level location information. This has no semantic impact and can safely be removed from a module.
This location information applies to the instructions physically following this instruction, up to the first occurrence of any of the following: **the next end of block, the next OpLine instruction, or the next [OpNoLine]**(https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpNoLine) instruction.

Reply:
After discussion with MrSidims, we decided that OpLine/OpNoLine should be ignored if OpExtInstImport for non-semantic debug information is encountered.  These imports are necessary to use DebugLine/DebugNoLine so the meaning is clear.

Reply:
Decided to not disable OpLine/OpNoLine if OpExtInstImport for non-semantic debug information is encountered.  We do not want a module that uses OpLine/OpNoLine to lose line information if it is linked with another module that has an OpExtInstImport for non-semantic debug information.   If OpLine and DebugLine are both active give DebugLine priority.